projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/KubernetesService.java

```diff
@@ -1480,7 +1480,7 @@ private V1Secret buildV1Secret(String name, Map<String, byte[]> data) {
                 deployment.setImageName(image); // TODO do we really need to return image_name?
             }
             var labels = cronJob.getSpec().getJobTemplate().getMetadata().getLabels();
-            if (labels.containsKey(JobLabel.VERSION.getValue())) {
+            if (labels != null && labels.containsKey(JobLabel.VERSION.getValue())) {
                 deployment.setGitCommitSha(labels.get(JobLabel.VERSION.getValue()));
             } else {
                 // Legacy approach to get version:
```

Comment:
can you add log for which we do not have labels expected. It could be even warning.

Reply:
I will. This particular one was the ecr-cred-helper.